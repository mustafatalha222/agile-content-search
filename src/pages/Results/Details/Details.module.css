.details {
  display: flex;
  flex-direction: column;
  padding: 1rem;
  background: #fff;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  margin-top: 1rem;
  max-width: 350px;
}

.imageWrapper {
  flex: 1;
}

.image {
  width: 100%;
  border-radius: 4px;
  object-fit: cover;
}

.url {
  font-size: 0.9rem;
  margin-bottom: 0.2rem;
}

.title {
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  display: inline;
}

.description {
  font-size: 1rem;
  line-height: 1.5;
  color: #4d5156;
}
