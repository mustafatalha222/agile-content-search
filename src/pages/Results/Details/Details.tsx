import styles from './Details.module.css'
import { IAnimal } from '../../../types/Animal'

type DetailsProps = {
  item: IAnimal | null
}

const Details = ({ item }: DetailsProps) => {
  if (!item) return

  return (
    <main className={styles.details}>
      <div className={styles.imageWrapper}>
        <img src={item.image} alt={item.title} className={styles.image} loading="lazy" />
      </div>
      <div className={styles.content}>
        <header>
          <div className={styles.url}>{item.url}</div>
          <h3 className={styles.title}>{item.title}</h3>
        </header>
        <section className={styles.description}>{item.description}</section>
      </div>
    </main>
  )
}

export default Details
