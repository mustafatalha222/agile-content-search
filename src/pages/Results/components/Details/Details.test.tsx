import { render } from '@testing-library/react'
import Details from '.'
import { mockAnimals } from '../../../../utils/test/constant'

describe('Details Component', () => {
  it('should render and match snapshot', () => {
    const { asFragment } = render(<Details item={mockAnimals[0]} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
