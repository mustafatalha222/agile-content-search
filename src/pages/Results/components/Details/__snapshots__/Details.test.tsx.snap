// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component should render and match snapshot 1`] = `
<DocumentFragment>
  <main
    class="details"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="lion"
        class="image"
        loading="lazy"
        src="https://example.com/animal.jpg"
      />
    </div>
    <div
      class="content"
    >
      <header>
        <div
          class="url"
        >
          https://example.com
        </div>
        <h3
          class="title"
        >
          lion
        </h3>
      </header>
      <section
        class="description"
      >
        This is an example description.
      </section>
    </div>
  </main>
</DocumentFragment>
`;
